### ISSUE TITLE
Universal backend: Fix ruby target

### ISSUE THREAD

**Issue by feeley** [gambit/gambit#233]
Estimated time: 10 hours

Not too far back the ruby backend was working fine, but some commit has broken it.  Figure out what is wrong and fix it!

```
% cat test.scm
(println 42)
% gsc/gsc -:=. -link -flat -target js test.scm
% wc test_.js test.js
     263     728    5608 test_.js
      38     108    1098 test.js
     301     836    6706 total
% cat test_.js test.js > prog.js
% node prog.js
42
% gsc/gsc -:=. -link -flat -target python test.scm
% wc test_.py test.py
     265     697    5530 test_.py
      46     125    1251 test.py
     311     822    6781 total
% cat test_.py test.py > prog.py
% python prog.py
42
% gsc/gsc -:=. -link -flat -target ruby test.scm
% wc test_.rb test.rb 
     298     669    5527 test_.rb
      40     118    1178 test.rb
     338     787    6705 total
% cat test_.rb test.rb > prog.rb
% ruby prog.rb
prog.rb:130:in `<main>': can't modify frozen NilClass (RuntimeError)
```

**Comment by eval-apply** [comment]
Perhaps this is due to an updated Ruby rather than a change in Gambit? 

https://bugs.ruby-lang.org/issues/8923

**Comment by feeley** [comment]
Seems like it is a problem with how functions for control-points are defined. They need to be Ruby "lambda" functions so that they can be extended to add function attributes (id, parent, fs, link).  For some reason they are currently being defined with the "def foo ... end" syntax... should be "foo = lambda do ... end".

**Comment by feeley** [comment]
Fixed by commit 2c9cbb7cccc9c039cba672a6b566c3b56707c788
